### PR TITLE
fix: replace SYSTEMROOT with SYSDIR

### LINKED ISSUE #8131: After upgrading to 24.13.2, Windows NSIS installer Failed to uninstall old application files when there is an app running
* **Electron-Builder Version**: 24.13.2

* **Node Version**: 18.14.0

* **Electron Version**: 29.1.1
* **Electron Type (current, beta, nightly)**: current

* **Target**: Windows

After upgrading to 24.13.2, when installing nsis installer when app is running, it shows below error "**xxxx cannot be closed. Please close it manually and click Retry to continue**"

electron-quick-start-is-running

Click **Retry** will keep showing the dialog again. When click **Cancel**, it shows "**old application files cannot be uninstalled**"
electron-quick-start-is-cannot-uninstall-old-app

**Versions before 24.13.2 do not have this issue**

When try previous version **24.13.1**,  It shows dialog **xxx is running, click OK to close**. After click **OK**, running app will close and install will continue.

electron-quick-start-13 1

### Step to reproduce
- upgrade electron-builder to 24.13.2 and create a NSIS installer
- install the installer and have the app running
- click the installer again to install or make another new installer to install

Seems like the fix in https://github.com/electron-userland/electron-builder/pull/8059 introduce this issue.  Have anyone run into the same issue?

Thx a million !

### PR DESCRIPTION
**Pull request by beyondkmp**
fix #8131

### CODE REVIEW

**Comment by mmaietta** [comment]
Interesting, nice catch!

Looks like a test is failing with the changes though when compiling via docker `TEST_FILES=protonTest pnpm test-linux`
```
FAIL  src/protonTest.ts (40.249 s)
  ✓ linux (7412 ms)
  ✕ win (18135 ms)
  ✕ win ia32 (13049 ms)

    Processing script file: "<stdin>" (UTF8)
    warning 6000: unknown variable/constant "%SYSTEMROOT%\System32\cmd.exe"" detected, ignoring (macro:FIND_PROCESS:5)
```

**Comment by beyondkmp** [comment]
@mmaietta  Fix it.

File: packages/app-builder-lib/templates/nsis/include/allowOnlyOneInstallerInstance.nsh [hunk 1]
```diff
@@ -91,7 +91,7 @@
           !ifdef INSTALL_MODE_PER_ALL_USERS
             nsExec::Exec `taskkill /f /im "${APP_EXECUTABLE_FILENAME}" /fi "PID ne $pid"`
           !else
-            nsExec::Exec `%SYSTEMROOT%\System32\cmd.exe /c taskkill /f /im "${APP_EXECUTABLE_FILENAME}" /fi "PID ne $pid" /fi "USERNAME eq %USERNAME%"`
+            nsExec::Exec `"$SYSDIR\cmd.exe" /c taskkill /f /im "${APP_EXECUTABLE_FILENAME}" /fi "PID ne $pid" /fi "USERNAME eq %USERNAME%"`
```
Comment (mmaietta) [inline packages/app-builder-lib/templates/nsis/include/allowOnlyOneInstallerInstance.nsh:94]: What is `$SYSDIR`? I don't see it when I get env vars in powershell or command prompt

Comment (beyondkmp) [reply packages/app-builder-lib/templates/nsis/include/allowOnlyOneInstallerInstance.nsh:94]: $SYSDIR is predefined in NSIS scripts. It's an NSIS var. 
https://nsis.sourceforge.io/Docs/Chapter4.html
image

Comment (mmaietta) [reply packages/app-builder-lib/templates/nsis/include/allowOnlyOneInstallerInstance.nsh:94]: Nice! Thank you for clarifying

**Pull request merged**